src/components/fields/edit/LocationIdEditor.jsx

```diff
@@ -20,67 +34,46 @@ export default function TreeViewInput(props) {
     ...rest
   } = props;
 
-  const label = useLabel(schema);
-  const description = useDescription(schema);
-  const showDescription = !minimalLabels && description;
-
-  const intl = useIntl();
-  const [expanded, setExpanded] = useState([]);
-  const [selected, setSelected] = useState([]);
-
-  function renderLevel(leaves) {
-    if (!leaves || leaves.length === 0) return null;
-    return leaves.map(leaf => (
-      <TreeItem key={leaf.id} nodeId={leaf.id} label={leaf.name}>
-        {leaf.locationID && renderLevel(leaf.locationID)}
-      </TreeItem>
-    ));
+  function getNumDescendents(targetChoice) {
+    let result = 0;
+    if (!targetChoice.locationID) return result;
+    const childCountsArr = targetChoice.locationID.map(child => {
+      return getNumDescendents(child);
+    });
+    result = targetChoice.locationID.length + childCountsArr.reduce((a, b) => a + b, 0);
+    return result;
   }
 
-  const handleToggle = (event, toggledNodes) => {
-    setExpanded(toggledNodes);
-  };
-
-  const handleSelect = (event, nodeIds) => {
-    if (!schema.multiselect) {
-      onChange(nodeIds);
-    } else {
-      onChange(nodeIds.filter(id => id !== 'IGNORE'));
-    }
-    setSelected(nodeIds);
-  };
+  function collapseChoices(choices, depth) {
+    const result = choices.map(choice => {
+      if (!choice.locationID) {
+        return { depth, name: ' '.repeat(depth) + choice.name + (getNumDescendents(choice) ? '(' + getNumDescendents(choice).toString() + ')' : ''), id: choice.id };
```

Comment:
this is too much for one line i think:
a) don't forget to run the formatter before pushing (`npm run format`)
b) defining some variables above the return would make it more legible i think
c) since we are doing the spacing in the css (on line 64) we could eliminate the spaces right?
d) dont forget about those template literals! they are much better than building strings this way 

```
const numberOfDescendants = getNumDescendants(choice);
const descendantsString = numberOfDescendants ? ` (${numberOfDescendants})` : '';
const name = choice.name + descendantsString;
return { depth, name, id: choice.id};
```